server/src/main/java/com/objectcomputing/checkins/services/file/FileServicesImpl.java

```diff
@@ -0,0 +1,226 @@
+package com.objectcomputing.checkins.services.file;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.InputStreamContent;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.model.File;
+import com.google.api.services.drive.model.FileList;
+import com.objectcomputing.checkins.notifications.email.EmailSender;
+import com.objectcomputing.checkins.services.checkindocument.CheckinDocument;
+import com.objectcomputing.checkins.services.checkindocument.CheckinDocumentServices;
+import com.objectcomputing.checkins.services.checkins.CheckIn;
+import com.objectcomputing.checkins.services.checkins.CheckInServices;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileServices;
+import com.objectcomputing.checkins.services.role.RoleType;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.multipart.CompletedFileUpload;
+import io.micronaut.security.utils.SecurityService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+import java.io.ByteArrayOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.time.LocalDate;
+import java.util.*;
+
+@Singleton
+public class FileServicesImpl implements FileServices {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FileServicesImpl.class);
+
+    private final GoogleDriveAccessor googleDriveAccessor;
+    private final EmailSender emailSender;
+    private final SecurityService securityService;
+    private final CheckInServices checkInServices;
+    private final CheckinDocumentServices checkinDocumentServices;
+    private final MemberProfileServices memberProfileServices;
+
+    public FileServicesImpl(GoogleDriveAccessor googleDriveAccessor, EmailSender emailSender,
+                            SecurityService securityService, CheckInServices checkInServices,
+                            CheckinDocumentServices checkinDocumentServices,
+                            MemberProfileServices memberProfileServices) {
+        this.googleDriveAccessor = googleDriveAccessor;
+        this.emailSender = emailSender;
+        this.securityService = securityService;
+        this.checkInServices = checkInServices;
+        this.checkinDocumentServices = checkinDocumentServices;
+        this.memberProfileServices = memberProfileServices;
+    }
+
+    @Override
+    public HttpResponse<?> findFiles(@Nullable UUID checkInID) {
+
+        Boolean isAdmin = securityService!=null ? securityService.hasRole(RoleType.Constants.ADMIN_ROLE) : false;
+        validate(checkInID == null && !isAdmin, "You are not authorized to perform this operation");
```

Comment:
From the DoD:

- [ ] find by id should only be allowed by:
  - [ ] ADMIN
  - [ ] the current PDL for the user whose check-in it is
  - [ ]  PDL for that check-in

I think you are missing those last two cases.

Reply:
updated in the latest commit.